ios/FluentUI/Vnext/PersonaButton/PersonaButton.swift

```diff
@@ -0,0 +1,273 @@
+        }
+        set {
+            avatarState.isOutOfOffice = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var isRingVisible: Bool {
+        get {
+            return avatarState.isRingVisible
+        }
+        set {
+            avatarState.isRingVisible = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var isTransparent: Bool {
+        get {
+            return avatarState.isTransparent
+        }
+        set {
+            avatarState.isTransparent = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var presence: MSFAvatarPresence {
+        get {
+            return avatarState.presence
+        }
+        set {
+            avatarState.presence = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var primaryText: String? {
+        get {
+            return avatarState.primaryText
+        }
+        set {
+            avatarState.primaryText = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var ringColor: UIColor? {
+        get {
+            return avatarState.ringColor
+        }
+        set {
+            avatarState.ringColor = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var secondaryText: String? {
+        get {
+            return avatarState.secondaryText
+        }
+        set {
+            avatarState.secondaryText = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var size: MSFAvatarSize {
+        get {
+            return avatarState.size
+        }
+        set {
+            avatarState.size = newValue
+            objectWillChange.send()
+        }
+    }
+
+    var style: MSFAvatarStyle {
+        get {
+            return avatarState.style
+        }
+        set {
+            avatarState.style = newValue
+            objectWillChange.send()
+        }
+    }
+
+    init(size: MSFPersonaButtonSize, avatarState: MSFAvatarState) {
+        self.badgeSize = size
+        self.avatarState = avatarState
+        self.tokens = MSFPersonaButtonTokens(size: size)
+
+        super.init()
+    }
+
+    private var avatarState: MSFAvatarState
+}
+
+public struct PersonaButton: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var tokens: MSFPersonaButtonTokens
+    @ObservedObject var state: MSFPersonaButtonViewStateImpl
+    let avatarView: AvatarView
+
+    public init(size: MSFPersonaButtonSize) {
+        self.avatarView = AvatarView(style: .default, size: size.avatarSize)
+
+        let state = MSFPersonaButtonViewStateImpl(size: size, avatarState: avatarView.state)
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        let action = state.onTapAction ?? {}
+        Button(action: action) {
+            VStack(spacing: 0) {
+                avatarView
```

Comment:
I think we don't need this anymore since you're adding the internal initializer for the Avatar.
I think we can create the avatar state inside the MSFPersonaButtonViewStateImpl, remove the avatar state param from the initializer and just pass the avatar state created here, right?

Reply:
Yep! This just didn't have that PR yet, but I'll merge it over